controller/pkg/extensions/processor.go

```diff
@@ -0,0 +1,13 @@
+package extensions
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ExtensionProcessor defines how to process extensions for a specific resource type.
+type ExtensionProcessor interface {
```

Comment:
Since this exists in the context of `extensions` package the name `ExtensionProcessor` stutters (This typically is detected by the linter, not sure why it wasn't).

I suggest to call this simply `Processor`, this way at the call site it's `extensions.Processor`. WDYT?

(not suggesting a change here through GH UI as that will require a change in other files as well).